diagnostics/collector.go

```diff
@@ -0,0 +1,250 @@
+		}
+		defer rows.Close()
+
+		filename := name + csvFileExt
+		w, err := writer.Create(filename)
+		if err != nil {
+			return nil, fmt.Errorf("new file for table %s - %w", name, err)
+		}
+		csvWriter := csv.NewWriter(w)
+		defer csvWriter.Flush()
+
+		first := true
+		for rows.Next() {
+			// write csv header row
+			if first {
+				first = false
+				descriptions := rows.FieldDescriptions()
+				if len(descriptions) == 0 {
+					return nil, errNoColumnsFound
+				}
+				cols := make([]string, len(descriptions))
+				for i, fd := range descriptions {
+					cols[i] = string(fd.Name)
+				}
+				if err := csvWriter.Write(cols); err != nil {
+					return nil, fmt.Errorf("write csv header for %s: %w", name, err)
+				}
+			}
+
+			// format and write data row
+			v, err := rows.Values()
+			if err != nil {
+				return nil, err
+			}
+			record := make([]string, len(v))
+			for i := range v {
+				record[i] = fmt.Sprintf("%v", v[i])
+			}
+			if err := csvWriter.Write(record); err != nil {
+				return nil, err
+			}
+		}
+		return nil, rows.Err()
+	}, db.WithContext(ctx), db.ReadOnly())
+	return err
+}
+
+func (c *Collector) writeTableCount(ctx context.Context, writer *zip.Writer, name string) error {
+	q := sq.Select("COUNT(*)").From(name)
+	return c.writeQueryContent(ctx, writer, name+"_count", q)
+}
+
+func (c *Collector) rangesStats(ctx context.Context, writer *zip.Writer) error {
+	var combinedErr error
+	repos, _, err := c.cataloger.ListRepositories(ctx, -1, "")
+	if err != nil {
+		// Cannot list repos, nothing to do..
+		combinedErr = multierror.Append(combinedErr, fmt.Errorf("listing repositories: %w", err))
+		return combinedErr
+	}
+
+	rangesFile, err := writer.Create("graveler-ranges")
+	if err != nil {
+		combinedErr = multierror.Append(combinedErr, fmt.Errorf("creating ranges file: %w", err))
+		return combinedErr
+	}
+
+	csvWriter := csv.NewWriter(rangesFile)
+	defer csvWriter.Flush()
+	if err := csvWriter.Write([]string{"repo", "type", "count(max:1000)"}); err != nil {
+		combinedErr = multierror.Append(combinedErr, fmt.Errorf("writing headers: %w", err))
+	}
+
+	for _, repo := range repos {
+		count := 0
+		counter := func(id string) error {
+			count++
+			return nil
+		}
+
+		if err := c.adapter.Walk(block.WalkOpts{
+			StorageNamespace: repo.StorageNamespace,
+			Prefix:           c.pyramidCfg.BlockStoragePrefix,
+		}, counter); err != nil {
+			combinedErr = multierror.Append(combinedErr, fmt.Errorf("listing ranges and meta-ranges: %w", err))
+		}
+		if err := csvWriter.Write([]string{repo.Name, "range", strconv.Itoa(count)}); err != nil {
+			combinedErr = multierror.Append(combinedErr, fmt.Errorf("writing meta-ranges for repo (%s): %w", repo.Name, err))
+		}
+	}
+
+	return combinedErr
+}
+
+func writeErrors(writer *zip.Writer, err error) error {
+	if err == nil {
+		return nil
+	}
+	combinedErr, ok := err.(*multierror.Error)
+	if !ok {
+		logging.Default().Error("error not of type multierror.Error: %w", err)
+	}
```

Comment:
Think you can remove the cast and just print the error - the multierror got a multiline format so the single print should work for any error